clients/src/main/java/org/apache/kafka/common/network/ChannelState.java

```diff
@@ -67,20 +68,23 @@
     // AUTHENTICATION_FAILED has a custom exception. For other states,
     // create a reusable `ChannelState` instance per-state.
     public static final ChannelState NOT_CONNECTED = new ChannelState(State.NOT_CONNECTED);
-    public static final ChannelState AUTHENTICATE = new ChannelState(State.AUTHENTICATE);
     public static final ChannelState READY = new ChannelState(State.READY);
     public static final ChannelState EXPIRED = new ChannelState(State.EXPIRED);
     public static final ChannelState FAILED_SEND = new ChannelState(State.FAILED_SEND);
     public static final ChannelState LOCAL_CLOSE = new ChannelState(State.LOCAL_CLOSE);
 
     private final State state;
     private final AuthenticationException exception;
+    private final SocketAddress remoteAddress;
+
     public ChannelState(State state) {
-        this(state, null);
+        this(state, null, null);
     }
-    public ChannelState(State state, AuthenticationException exception) {
+
+    public ChannelState(State state, AuthenticationException exception, SocketAddress remoteAddress) {
```

Comment:
It feels odd to store a `SocketAddress` in this class. Perhaps we could store a String `channelDescription` or something along those lines to contain any additional channel details that may be useful?

Reply:
Sorry, I wasn't clear with the earlier comment. This constructor was being used specifically for `AUTHENTICATION_FAILED`, hence the `AuthenticationException`. So adding remote address as a `SocketAddress` just for authentication failures looked odd since the remote address is also associated with various other states. So something like `channelDescription` felt more appropriate to add additional context for an authentication failure. Now that you have added it to other states as well, `remoteAddress` is fine and you can have it as `String` or `SocketAddress`. I leave it up to you whether you want to revert.